src/libply/provider/interval.c

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright  Namhyung Kim <namhyung@gmail.com>
+ *
+ * SPDX-License-Identifier: GPL-2.0
+ */
+
+#include <assert.h>
+#include <errno.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include <ply/ply.h>
+#include <ply/internal.h>
+
+
+struct interval_data {
+	unsigned long long interval;
+	int fd;
+};
+
+static int interval_sym_alloc(struct ply_probe *pb, struct node *n)
+{
+	return -ENOENT;
+}
+
+static int interval_probe(struct ply_probe *pb)
+{
+	char *unit;
+	char *num;
+	unsigned long long intvl;
+	struct interval_data *data;
+
+	unit = strchr(pb->probe, ':');
+	if (unit == NULL) {
+		_e("interval doesn't have unit: %s\n", pb->probe);
+		return -1;
+	}
+	unit++;
+
+	num = strchr(unit, ':');
+	if (num == NULL) {
+		_e("interval doesn't have number: %s\n", pb->probe);
+		return -1;
+	}
+	num++;
+
+	intvl = strtoull(num, NULL, 0);
+
+	switch (*unit) {
+	case 's':
+		intvl *= 1e9;  /* seconds */
+		break;
+	case 'm':
+		if (unit[1] == 's')
+			intvl *= 1e6;  /* milli-seconds */
+		else if (unit[1] == ':')
+			intvl *= 60 * 1e9;  /* minutes */
+		else {
+			_e("invalid time unit: %s\n", unit);
+			return -1;
+		}
+		break;
+	case 'u':
+		if (unit[1] != 's') {
+			_e("invalid time unit: %s\n", unit);
+			return -1;
+		}
+		intvl *= 1e3;  /* micro-seconds */
+		break;
+	case 'n':
+		if (unit[1] != 's') {
+			_e("invalid time unit: %s\n", unit);
+			return -1;
+		}
+		/* nano-seconds */
+		break;
+	default:
+		_e("unsupported time unit: %s\n", unit);
+		return -1;
+	}
```

Comment:
Something like `interval:ss:1` or `interval:nss:100` are accepted by this check, right ?

Reply:
Thanks for your review, will change.